assignment3/wordSearch.py

```diff
@@ -0,0 +1,87 @@
+from dictionary import Dictionary
+from collections import deque
+
+
+class GridWord:
+    """ A class that represents a path in the grid and a corresponding word.
+
+    Fields:
+        _last_position:    a tuple of length 2 that represents the coordinates
+                           of the last position in the path;
+        _visited_position: a set of tuples that contains all coordinates
+                           in this path;
+        _word:             a string, the word that corresponds to the path.
+    """
+    def __init__(self, last = None, visited = None, word = ''):
+        self._last_position = last
+        if visited is None:
+            self._visited_positions = set()
+        else:
+            self._visited_positions = visited
+        self._word = word
+
+    def getLastPosition(self):
+        return self._last_position
+
+    def getWord(self):
+        return self._word
+
+    def newPosition(self, i, j, symbol):
```

Comment:
Maybe `appendPosition` would be more appropriate.
Naming something like `moveTo(i, j, symbol)` might also make sense for this specific case.